### ISSUE TITLE
Screenreader Issue - Modal Windows broken with VO and Mobile Safari

### ISSUE THREAD

**Issue by James-Green** [w3c/wai-quick-start#295]
The pop-up modals in the Writing - Headings example and Developing – Space Trainers example appear broken in Mobile Safari – they are crashing the browser pretty often with VO open and when they don’t crash, they start scrolled to the bottom visually (and focus stays on the modal launching link, when you move from it you end up near the bottom of the modal’s content). 

Moving down with VO from this starting place near the bottom of the modal content ends up bottoming out instead of wrapping to the top of the modal.  It's only bottoming out though because the modal is the last thing on the page, otherwise it would move on to content behind the modal as it does when going up... Moving up from that starting place near the bottom of the modal content does not wrap to the bottom of the modal when you get to the top, it goes to the content behind the modal starting at the bottom of the page, because the modal is at the bottom of the page in the code order but focus is not trapped.  

Touching outside the modal allows the user to scroll the content behind. 

 Suggested enhancements:
* trap focus inside modal 
* wrap focus inside modal
* set focus to text right after H1 (since H1 is read as the aria-label)
* prevent clicks/touches outside modal from having any effect.

![screen shot 2015-09-30 at 10 35 19 pm](https://cloud.githubusercontent.com/assets/13589074/10212334/492327aa-67c3-11e5-88a3-a5ccbc60fd68.png)

**Comment by iadawn** [comment]
I have tested this using JAWS (16 with IE 11) and VO on iPad (iOS 8.2). Couple of things I have found:

• On iPad, it is not possible to scroll behind the modal pop-up
• On iPad, it is possible to activate links to the side of the modal pop-up
• On iPad, it is possible to move the VO virtual cursor outside of the modal pop-up
• It is not possible to move keyboard focus outside of the modal pop-up
• In IE, it is not possible to click on links outside of the modal pop-up
• In Firefox (and probably IE), it is possible to scroll the background window
• In JAWS, virtual cursor can move outside of modal pop-up, or at least it can read out of the window

Some thoughts:

• I don't think it is possible to trap the virtual cursor or reading of a screen reader in a window of this sort. Unless addding the `role='dialog'` is intended to be used to do something like this, which I think it might be. Unfotunetely, it doesn't appear that screen readers handle this well.
• Scolling behind the modal, I don't see as a huge problem since when the dialog is closed, focus returns to the activating link.
• Clicking outside the modal, ideally this should be addressed although I can only replicate the behavior on the iPad (and probably iPhone) so this may be low priority.
• Note, in Safari and Chrome if I click outside of the modal, I can't activate elements but focus does shift out of the modal window and then cycles through the background page. This should be fixed.
